### ISSUE TITLE
fix string interpolation in class names during class-name-format validation

### ISSUE THREAD

**Issue by sormy** [sasstools/sass-lint#988]
> What version of Sass Lint are you using?

1.9.1

> Please include any relevant parts of your configuration

```
rules:
  class-name-format: [2, { convention: strictbem } ]
```

> What did you do? Please include the actual source code causing the issue.

```scss
$primary-class: 'bla-bla-bla';

.#{$primary-class}--small {
  ...

  &__child {
    ...
  }
}
```

> What did you expect to happen?

No error.

> What actually happened? Please include any error messages given to you by Sass Lint.

[sass-lint] Class '.--small' should be written in BEM (Block Element Modifier) format
False positive error. Class name should be 'bla-bla-bla--small' and should pass BEM class name validation.

> If you're using a IDE plugin have you tried the CLI too?

IDE Plugin, VS Code

**Comment by DanPurdy** [comment]
We don't compile your Sass to check for things like this. The only option we have is to ignore interpolation and make sure that the remaining text passes a sort of validation here.

**Comment by DanPurdy** [comment]
Released in 1.12.0
